### DISCUSSION TITLE
How to query data from the timescaledb extension using the javascript client?

### DISCUSSION THREAD

**Discussion by jeremytenjo** [supabase/supabase#18245] [Questions] [answered]
How do we run this `timescaledb` query using the javascript client?

```sql
select
    time_bucket('1 hour', time) AS hour,
    avg(temperature) AS average_temperature
from
    temperatures
where
    sensor_id = 1
    and time > NOW() - interval '1 hour'
group by
    hour;
```

https://supabase.com/docs/guides/database/extensions/timescaledb?database-method=dashboard
https://supabase.com/docs/reference/javascript/select

Thanks!

**Comment by GaryAustin1** [comment, marked as answer]
The clients do not support group by.  Also you cannot include a function in the select.
You are better off using an rpc call to a postgres function and run your above query returning a table.

**Reply by jeremytenjo** [reply]
I see, thanks @GaryAustin1 !
